### DISCUSSION TITLE
I want to know the upstream(source_workload) of ingress-gateway

### DISCUSSION THREAD

**Discussion by seun9su** [istio/istio#47640] [Q&A]
Currently, our istio configuration is as follows:

user -> NLB -> ingress-gateway POD -> Service Application

In the current structure, all traffic passes through the ingress-gateway, so if you check the istio_requests_total metric, source_workload is couted to the ingress-gateway.

ex) 
<img width="2047" alt="image" src="https://github.com/istio/istio/assets/22903695/685770cc-2ae6-4656-902d-c33d1bf7bde7">

In the above capture, since calls between each service also pass through the ingress-gateway, source_workload is counted as "internal-ingressgateway".

In the above capture, since calls between each service also pass through the ingress-gateway, source_workload is counted as internal-ingressgateway.

For example, if you call test-application1 -> test-application2, when you check the istio_requests_total metric, I want the source_workload to be test-application1 and the destination_app to be test-application2. In the current structure, as mentioned above, all source_workloads are recorded as ingressgateway.

When making an internal call, if request through mesh (sidecar communication), it will go in the direction I want, but since mesh communication only supports http, everything must pass through the ingress-gateway.

Is it possible to set it up with just istio settings?

**Comment by rootsongjc** [comment]
Have you tried using ServiceEntry to define these two non-HTTP services? This way, their traffic wouldn't go through the Ingress Gateway.

**Reply by seun9su** [reply]
Because there is an NLB in front of the ingressgateway, the ingressgateway must be passed for TLS offloading. If the source_workload in istio_requests_total is an ingressgateway, I would like to know the upstream source_workload, excluding that source_workload. Do you know how?

**Reply by howardjohn** [reply]
If you want to know the source from multiple hops away you would need to encode that in the HTTP headers. XFF is a standard way to do that (https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/), but you could manually do it in a custom way with virtual service adding a header as well
